### ISSUE TITLE
[Feature Request] Add support for code snippets

### ISSUE THREAD

**Issue by leosolid** [lemnos/tt#11]
I'd to practice typing code as well. Not sure how easy this would be to add though...

**Comment by lemnos** [comment]
I'm not sure a dedicated feature for this makes sense. What do you imagine the implementation would like? It is already possible to use tt with code (or indeed any arbitrary text). For code you would probably want to use the -raw flag and feed in small chunks at a time. E.G `head -n 20 code.c|tt -raw`.

**Comment by jmonroynieto** [comment]
I agree with this. I forked this, and it is working. Raw mode now captures tabs, but they are rendered as spaces.
https://github.com/jmonroynieto/tt_withTab/commit/e080d33a9cb72e625cda9857f3e20d6a7ee3cae9


//TODO: render as multiple characters such as `›··`, `▸··`, or `⍿··`. Possibly accomplished by special handling of redraw. Now that typer.go::Start() knows about raw mode, this could be accomplished.
//TODO: Handle the cursor correctly when masking tabs in raw mode.
//TODO: Create a special instance of reflow for raw mode where four spaces make a tab character. tt.go@L404

I'll put in a PR when it is complete. Anyone interested in push access, let me know.

**Comment by jmonroynieto** [comment]
Also, for those interested, I wrote a simple tool to capture quote/code snippets interactively. See [quoteadder](https://github.com/jmonroynieto/cliWorkflow_tk/tree/main/quoteadder).

Would the owners be willing to merge this as a command to tt?

**Comment by jmonroynieto** [comment]
The suggested solution only changes 4 spaces to tabs, other amounts are unchanged. This is a bit bothersome for go code that has been formatted since some instances like struct definition can have odd spacing to make it visually clean. 

Making it "squish" sequential spaces after changing groups of 4 to tabs would be trivial and perhaps more useful, let me know.
